File: Cloudy_Self-Checkout_POS_System-main/src/main/java/com/Cloudy/Cloudy_Self_Checkout_POS/POJO/Purchase.java
package com.Cloudy.Cloudy_Self_Checkout_POS.POJO;

import jakarta.persistence.*;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicInsert;
import org.hibernate.annotations.DynamicUpdate;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@Entity
@DynamicUpdate
@DynamicInsert
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "purchase")
public class Purchase {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long purchaseId;

    @Column(nullable = false,updatable = false)
    private String guid;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(precision = 10, scale = 2)
    private BigDecimal taxAmount;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal originalAmount;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal totalAmount;

    @Column( precision = 10, scale = 2)
    private BigDecimal discountAmount;

    @Column( precision = 10, scale = 2)
    private BigDecimal couponDiscount;

    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal finalAmount;

    @Column(nullable = false)
    private String status; // PENDING, COMPLETED, FAILED (only final after all retries)

    @ManyToOne
    @JoinColumn(name = "coupon_id")
    private Coupon coupon; // Coupon applied

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdOn;

    @Column(nullable = false)
    private String Currency;

//    @Column(unique = true)
//    private String transactionId;

}

File: cloudy_pos_frontend-main/src/app/customer/Model/Purchase.ts
export interface PurchaseHistory {
  purchaseId: number;
  guid: string;
  userId: number;
  username: string;
  taxAmount: number;
  originalAmount: number;
  totalAmount: number;
  discountAmount: number;
  couponDiscount: number;
  finalAmount: number;
  status: string;
  couponId: number | null;
  createdOn: string;
  currency: string | null;
  items: PurchaseItem[];
  transaction: Transaction | null;
  voucherPdfUrl: string | null;
}

export interface PurchaseItem {
  purchaseItemId: number;
  guid: string;
  purchaseId: number;
  productId: number;
  quantity: number;
  price: number;
  subTotal: number;
}

export interface Transaction {
  transactionId: number;
  guid: string;
  purchaseId: number;
  paymentId: string;
  amount: number;
  status: string;
  paymentMethod: string;
  transactionDate: string;
  payerId: string | null;
  createdOn: string;
  failureReason: string | null;
}
